test/ssh_authorized_keys_test.rb

```diff
@@ -29,69 +31,78 @@
   let(:invalid_key_spec) { double("invalid_key") }
   let(:home) { "/home/user" }
   let(:key) { double("key") }
+  let(:keys) { [key] }
 
-  before do
-    allow(Yast::Users::SSHAuthorizedKey).to receive(:build_from)
-      .with(valid_key_spec)
-      .and_return(key)
-    allow(Yast::Users::SSHAuthorizedKey).to receive(:build_from)
-      .with(invalid_key_spec)
-      .and_return(nil)
-    ssh_authorized_keys.main
-  end
+  before { subject.main }
 
   describe "#import_keys" do
-    context "given an array which contains some valid keys specifications" do
-      let(:keys) { [valid_key_spec, invalid_key_spec] }
+    it "imports keys into the keyring" do
+      expect(subject.keyring).to receive(:add_keys).with(home, keys)
+        .and_return(keys)
+      subject.import_keys(home, keys)
+    end
 
-      it "returns true" do
-        expect(subject.import_keys(home, keys)).to eq(true)
-      end
+    it "returns true if some key was imported" do
+      allow(subject.keyring).to receive(:add_keys).with(home, keys)
+        .and_return(keys)
+      expect(subject.import_keys(home, keys)).to eq(true)
+    end
 
-      it "registers given keys" do
-        subject.import_keys(home, keys)
-        expect(subject.keyring[home]).to eq([key])
-      end
+    it "returns false if no key was imported" do
+      allow(subject.keyring).to receive(:add_keys).with(home, keys)
+        .and_return(keys)
+      expect(subject.import_keys(home, keys)).to eq(true)
     end
+  end
 
-    context "given an array which does not contain valid keys specifications" do
-      let(:keys) { [invalid_key_spec] }
+  describe "#export_keys" do
+    context "when some key was added" do
+      before { subject.import_keys(home, keys) }
 
-      it "returns false" do
-        expect(subject.import_keys(home, keys)).to eq(false)
+      it "returns an array with added keys" do
+        expect(subject.export_keys(home)).to eq(keys)
       end
+    end
 
-      it "does not register any key" do
-        subject.import_keys(home, keys)
-        expect(subject.keyring[home]).to be_empty
+    context "when no key was added" do
+      it "returns an empty array" do
+        expect(subject.export_keys(home)).to eq([])
       end
     end
   end
 
-  describe "#export_keys" do
-    let(:home) { "/home/user" }
-
-    context "when some key was registered for the given home" do
-      let(:key) do
-        double("key", options: "tunnel=0", keytype: "ssh-rsa", content: "123ABC",
-          comment: "user@example.net")
+  describe "#write_keys" do
+    context "when home directory does not exists" do
+      it "shows an error message" do
+        allow(subject.keyring).to receive(:write_keys)
+          .and_raise(Yast::Users::SSHAuthorizedKeyring::HomeDoesNotExist)
+        expect(Yast::Report).to receive(:Warning)
+          .with("Home directory '#{home}' does not exist\n" \
```

Comment:
I'd be less strict about testing exact error messages, e.g. something like `with(/'#{home}' does not exist/)` should be enough. Shorter condition and smaller chance to break a test after fixing the message.

More over it does not add any value, you probably did copy&paste from the code. If there is a typo or a grammar issue it will be here as well.